### ISSUE TITLE
Supporting renderer from ArcGIS.com Feature layer item

### ISSUE THREAD

**Issue by bviswanathan** [Esri/esri-leaflet-renderers#51]
There are 2 types of feature services:
- Hosted feature services: You have a layer in ArcGIS Desktop or any JS API app, set the renderer on it and publish it using your ArcGIS.com account. Renderer is set on the service and the plugin renders it fine.
- Feature services: You have published a layer already to your server and want to add that layer to ArcGIS.com, apply the renderer (using smart mapping) in ArcGIS.com and save the layer to your user account. In this case, the renderer set on the item takes precedence over the renderer available from the service. I am able to access the renderer from the item, but Esri Leaflet Renderer defaults to the one in the service.

Question: Is there a way that I can force the Esri Leaflet Renderer’s layer.metadata call to use my own renderer instead of one from the service?

**Comment by jgravois** [comment]
the short answer is *no*.

i've always been aware that raw RESTful services have symbology defined (either with or without smart mapping) and that it was possible to override renderers for a layer and save the modification within Webmap JSON, but the third option you describe (new item in AGOL for the layer itself with custom symbology defined for someone else's service) is something i heard about for the first time the other day.

i have to assume that the ArcGIS Online/Portal REST API spec allows you to request JSON defining the item and that its something we could parse in this plugin, but help and samples from you would be incredibly helpful in order to implement the enhancement.

**Comment by patrickarlt** [comment]
> i have to assume that the ArcGIS Online/Portal REST API spec allows you to request JSON defining the item

There is actually no way (given a service url) to figure our the item URL. So the scenario that @bviswanathan wants (pass an service URL, figure out item ID, use item renderer) actually can't happen.

However this COULD be wrapped up in another plugin to expose an API like this:

```js
L.esri.FeatureLayer({
  id: 'ARCGIS Online Item ID'
}).addTo(map);
```

Then we know the item id so we can: 

1. Get the item metadata
2. This will have the service URL
3. We can check the `/data` end point on the item to see if a renderer is stored there
4. Use the appropriate renderer

Since this behavior goes beyond the scope of both Esri Leaflet (rendering the services) and Esri Leaflet Renderers (converting ArcGIS renderers into style functions) it should probably be a separate function.

**Comment by jwasilgeo** [comment]
Closing. Please refer to the linked (and closed) issues Esri/esri-leaflet#570 and ynunokawa/L.esri.WebMap#32.
